src/uos/adapters/outbound/http.py

```diff
@@ -0,0 +1,376 @@
+# Copyright 2021 - 2025 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Outbound HTTP calls"""
+
+import logging
+from typing import Any
+from uuid import UUID
+
+import httpx
+from ghga_service_commons.utils.utc_dates import UTCDatetime
+from jwcrypto import jwk
+from pydantic import UUID4, Field, HttpUrl, SecretStr
+from pydantic_settings import BaseSettings
+
+from uos.core.models import (
+    BaseWorkOrderToken,
+    ChangeFileBoxWorkOrder,
+    CreateFileBoxWorkOrder,
+    UploadGrant,
+    ViewFileBoxWorkOrder,
+)
+from uos.core.tokens import sign_work_order_token
+from uos.ports.outbound.http import AccessClientPort, FileBoxClientPort
+
+TIMEOUT = 60
+
+log = logging.getLogger(__name__)
+
+
+class AccessApiConfig(BaseSettings):
+    """Config parameters for managing upload access grants."""
+
+    access_url: HttpUrl = Field(
+        ...,
+        description="URL pointing to the internal access API.",
+        examples=["http://127.0.0.1/access"],
+    )
+
+
+class FileBoxClientConfig(BaseSettings):
+    """Config parameters for interacting with the service owning FileUploadBoxes."""
+
+    file_box_service_url: HttpUrl = Field(
```

Comment:
This could still be called `ucs_service_url` so that the devops people know what to configure.

My point was that only the FileBoxClient adapter itself needs to know that it's the UCS, not the user of the FlieBoxClient, therefore it should not be called UCSClient.